### PR TITLE
Homework3

### PR DESCRIPTION
**Pull request by dawidjozefczak**

### CODE REVIEW

File: module3/homework/passwordCheck/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,47 @@
...
+        return "Password needs at least one special character";
+    case PasswordNeedsAtLeastOneUppercaseLetter:
+        return "Password needs at least one uppercase letter";
+    case PasswordsDoesNotMatch:
+        return "Password does not match";        
+    }
+    return "Unknown error";
+}
+
+bool doesPasswordsMatch(const std::string& lhs, const std::string& rhs){
```
Comment (DamiXioR) [inline module3/homework/passwordCheck/validation.cpp:22]: in my opinion code would be more clear when you call variables as pass* etc.

File: module3/homework/passwordCheck/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,47 @@
...
+    return lhs == rhs;
+}
+
+ErrorCode checkPasswordRules(const std::string& str){
+    if(str.length() < MinPasswordLength){
+        return PasswordNeedsAtLeastNineCharacters;
+    }
+    else if(std::none_of(str.begin(), str.end(), ::isdigit)){
```
Comment (DamiXioR) [inline module3/homework/passwordCheck/validation.cpp:30]: it's easier to understand code when you use if with every condition

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#include "vectorFunctions.hpp"
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int count){
+    std::vector<std::shared_ptr<int>> vec;
+    vec.reserve(count);
```
Comment (DamiXioR) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:6]: Looka good when You check if count is bigger than 1

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+void print(const std::vector<std::shared_ptr<int>>& vec){
+    for(const auto& el : vec){
+        std::cout << *el << " ";
+    }
+    std::cout << "\n";
+}
+
+void add10(const std::vector<std::shared_ptr<int>>& vec){
+    for(const auto& el : vec){
+        if(el != nullptr){
```
Comment (DamiXioR) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:22]: just check if (el)

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            *el += 10;
+        }
+    }
+}
+
+void sub10(int* const ptr){
+    if(ptr != nullptr){
```
Comment (DamiXioR) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:29]: if (ptr) is enough

File: module3/homework/passwordCheck/validation.cpp [hunk 6]
```diff
@@ -1,2 +1,47 @@
...
+    }
+    return "Unknown error";
+}
+
+bool doesPasswordsMatch(const std::string& lhs, const std::string& rhs){
+    return lhs == rhs;
+}
+
+ErrorCode checkPasswordRules(const std::string& str){
+    if(str.length() < MinPasswordLength){
```
Comment (kyrtaq) [inline module3/homework/passwordCheck/validation.cpp:27]: 👍 for not using magic numbers

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 3 above)
Comment (kyrtaq) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:6]: check if count is not negative

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 4 above)
Comment (kyrtaq) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:22]: if(el) is enough

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 5 above)
Comment (kyrtaq) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:29]: if(ptr) is enough

File: module3/homework/passwordCheck/validation.hpp [hunk 7]
```diff
@@ -1,2 +1,18 @@
-// TODO: I'm empty :) Put enum and function headers here.
-// Don't forget the header guard
\ No newline at end of file
+#pragma once
+#include <string>
+
+constexpr size_t MinPasswordLength = 9;
```
Comment (nauka-programowania-MA) [inline module3/homework/passwordCheck/validation.hpp:4]: This variable is use only in one function. Please move it here.

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 4 above)
Comment (nauka-programowania-MA) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:22]: `if (el)` is enough

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 5 above)
Comment (nauka-programowania-MA) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:29]: `if (el)`

**Comment by nauka-programowania-MA** [comment]
{
"requirement_ids": [1995, 1996, 1119],
"points": [7, 4, 2]
}

**Comment by ziobron** [comment]
+5 XP Podstawy#3 – vectorOfSharedPointers (1998)
Without bonus for delivering before deadline ;)

{ "requirement_ids": [1998], "points": [5] }

**Pull request closed without merging**